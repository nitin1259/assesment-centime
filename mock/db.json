{
  "records": [
    {
      "id": 1,
      "title": "Extra bonus",
      "from_party": "BOA",
      "to_party": "Benz M",
      "amount": 34000
    },
    {
      "id": 2,
      "title": "hostility",
      "from_party": "GOldMan",
      "to_party": "Centime",
      "amount": 41000
    },
    {
      "id": 3,
      "title": "Same as Perks",
      "from_party": "Amazon",
      "to_party": "Benz",
      "amount": 21000
    },
    {
      "id": 4,
      "title": "Timber times",
      "from_party": "Benz",
      "to_party": "Centime",
      "amount": 66000
    },
    {
      "id": 5,
      "title": "Extra bonus",
      "from_party": "HDFC",
      "to_party": "Ferrari",
      "amount": 33000
    }
  ]
}