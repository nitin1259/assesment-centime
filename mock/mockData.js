const records = [
  {
    id: 1,
    title: "Extra bonus",
    from_party: "BOA",
    to_party: "Centime",
    amount: 34000,
  },
  {
    id: 2,
    title: "hostility",
    from_party: "Centime",
    to_party: "Goldman Sach",
    amount: 41000,
  },
  {
    id: 3,
    title: "Same as Perks",
    from_party: "Centime",
    to_party: "Antaric",
    amount: 21000,
  },
  {
    id: 4,
    title: "Timber times",
    from_party: "Benz",
    to_party: "Centime",
    amount: 66000,
  },
  {
    id: 5,
    title: "Extra bonus",
    from_party: "Centime",
    to_party: "Ferrari",
    amount: 13000,
  },
  {
    id: 6,
    title: "Funding",
    from_party: "Rera",
    to_party: "Centime",
    amount: 30000,
  },
  {
    id: 7,
    title: "Round1",
    from_party: "Centime",
    to_party: "Ferrari",
    amount: 3000,
  },
  {
    id: 8,
    title: "Extra bonus",
    from_party: "AON",
    to_party: "Centime",
    amount: 22000,
  },
];

// Using CommonJS style export so we can consume via Node (without using Babel-node)
module.exports = {
  records,
};
