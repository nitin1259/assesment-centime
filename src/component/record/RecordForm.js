import React from "react";
import PropTypes from "prop-types";
import TextInput from "../common/TextInput";
// import SelectInput from "../common/SelectInput";

function RecordForm({ record, onSave, errors, onChange, saving = false }) {
  return (
    <form onSubmit={onSave}>
      {/* <h2>{record.id ? "Edit" : "Add "} Record</h2> */}
      {errors && errors.onSave && (
        <div className="alert alert-danger" role="alert">
          {errors.onSave}
        </div>
      )}
      <TextInput
        name="title"
        label="Title"
        value={record.title}
        onChange={onChange}
        error={errors && errors.title}
      />

      <TextInput
        name="from_party"
        label="From"
        value={record.from_party}
        onChange={onChange}
        error={errors && errors.from_party}
      />

      <TextInput
        name="to_party"
        label="To"
        value={record.to_party}
        onChange={onChange}
        error={errors && errors.to_party}
      />

      <TextInput
        name="amount"
        label="Amount"
        value={record.amount}
        onChange={onChange}
        error={errors && errors.amount}
      />

      <button type="submit" disabled={saving} className="btn btn-primary">
        {saving ? "Saving..." : "Save"}
      </button>
    </form>
  );
}

RecordForm.propTypes = {
  onSave: PropTypes.func.isRequired,
  record: PropTypes.object.isRequired,
  errors: PropTypes.object,
  onChange: PropTypes.func.isRequired,
  saving: PropTypes.bool.isRequired,
};

export default RecordForm;
