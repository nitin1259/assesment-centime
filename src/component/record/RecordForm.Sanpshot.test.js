import React from "react";
import RecordForm from "./RecordForm";
import renderer from "react-test-renderer";
import { records } from "../../../mock/mockData";

it("sets submit button label 'Saving...' when saving is true", () => {
  const tree = renderer.create(
    <RecordForm
      record={records[0]}
      onSave={jest.fn()}
      onChange={jest.fn()}
      saving
    />
  );

  expect(tree).toMatchSnapshot();
});

it("sets submit button label 'Save' when saving is false", () => {
  const tree = renderer.create(
    <RecordForm
      record={records[0]}
      onSave={jest.fn()}
      onChange={jest.fn()}
      saving={false}
    />
  );

  expect(tree).toMatchSnapshot();
});
