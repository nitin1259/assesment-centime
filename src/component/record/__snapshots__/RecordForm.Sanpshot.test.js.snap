// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sets submit button label 'Save' when saving is false 1`] = `
<form
  className="form-class"
  onSubmit={[MockFunction]}
>
  <h2>
    Edit
     Record
  </h2>
  <div
    className="form-group"
  >
    <label
      htmlFor="title"
    >
      Title
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="title"
        onChange={[MockFunction]}
        type="text"
        value="Extra bonus"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="from_party"
    >
      From
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="from_party"
        onChange={[MockFunction]}
        type="text"
        value="BOA"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="to_party"
    >
      To
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="to_party"
        onChange={[MockFunction]}
        type="text"
        value="Benz"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="amount"
    >
      Amount
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="amount"
        onChange={[MockFunction]}
        type="text"
        value={34000}
      />
    </div>
  </div>
  <button
    className="btn btn-primary"
    disabled={false}
    type="submit"
  >
    Save
  </button>
</form>
`;

exports[`sets submit button label 'Saving...' when saving is true 1`] = `
<form
  className="form-class"
  onSubmit={[MockFunction]}
>
  <h2>
    Edit
     Record
  </h2>
  <div
    className="form-group"
  >
    <label
      htmlFor="title"
    >
      Title
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="title"
        onChange={[MockFunction]}
        type="text"
        value="Extra bonus"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="from_party"
    >
      From
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="from_party"
        onChange={[MockFunction]}
        type="text"
        value="BOA"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="to_party"
    >
      To
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="to_party"
        onChange={[MockFunction]}
        type="text"
        value="Benz"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="amount"
    >
      Amount
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="amount"
        onChange={[MockFunction]}
        type="text"
        value={34000}
      />
    </div>
  </div>
  <button
    className="btn btn-primary"
    disabled={true}
    type="submit"
  >
    Saving...
  </button>
</form>
`;
